standard/tier/tier_utils.lua

```diff
@@ -0,0 +1,191 @@
+---@param input string|integer|nil
+---@return string|integer|nil
+function Tier.toIdentifier(input)
+	if String.isEmpty(input) then
+		return
+	end
+
+	return tonumber(input)
+		or string.lower(input):gsub(' ', '')
+end
+
+--- Retrieves the raw data for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@return table?, table?
+function Tier._raw(tier, tierType)
+	return (TierData.tiers or {})[Tier.toIdentifier(tier)],
+		(TierData.tierTypes or {})[Tier.toIdentifier(tierType)]
+end
+
+--- Checks if a valid (tier, tierType) tuple is provided
+---@param tier integer
+---@param tierType string?
+---@return boolean
+function Tier.isValid(tier, tierType)
+	local tierData, tierTypeData = Tier._raw(tier, tierType)
+
+	if not tierData then return false end
+
+	if not tierTypeData and String.isNotEmpty(tierType) then
+		return false
+	end
+
+	return true
+end
+
+--- Converts input to (storage) values for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@return integer?, string|integer|nil
+function Tier.toValue(tier, tierType)
+	local tierData, tierTypeData = Tier._raw(tier, tierType)
+
+	return (tierData or {}).value, (tierTypeData or {}).value
+end
+
+--- Converts input to displayNames for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@return string?, string?
+function Tier.toName(tier, tierType)
+	local tierData, tierTypeData = Tier._raw(tier, tierType)
+
+	return (tierData or {}).name, (tierTypeData or {}).name
+end
+
+--- Converts input to short names for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@return string?, string?
+function Tier.toShortName(tier, tierType)
+	local tierData, tierTypeData = Tier._raw(tier, tierType)
+
+	return (tierData or {}).short, (tierTypeData or {}).short
+end
+
+--- Converts input to categories for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@return string?, string?
+function Tier.toCategory(tier, tierType)
+	local tierData, tierTypeData = Tier._raw(tier, tierType)
+
+	return (tierData or {}).category, (tierTypeData or {}).category
+end
+
+--- Converts input to a sort value for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@return string
+function Tier.toSortValue(tier, tierType)
+	local tierData, tierTypeData = Tier._raw(tier, tierType)
+
+	return (tierData or {}).sort .. ((tierTypeData or {}).sort or '')
+end
+
+--- Parses queryData to be processable for other Tier functions
+--- overwritable on a per wiki basis if additional data needs to be passed
+---@param queryData table
+---@return string?, string?, table
+function Tier.parseFromQueryData(queryData)
+	return queryData.liquipediatier, queryData.liquipediatiertype, {}
+end
+
+--- Builds the display for a given (tier, tierType) tuple
+---@param tier integer
+---@param tierType string?
+---@param options table?
```

Comment:
kinda hard to do that here because wikis might add additional options in here that commons does not expect
is there a way to list some options but also tell it that there might be more?

Reply:
Ok